Dual region pages update for clarity (#4342)
camunda/camunda-docs: docs/self-managed/operational-guides/multi-region/dual-region-ops.md

## Introduction

This operational blueprint procedure is a step-by-step guide on how to restore operations in the case of a total region failure. It explains how to temporarily restore functionality in the surviving region and how to ultimately do a full recovery to restore the dual-region setup. The operational procedure builds on top of the [dual-region AWS setup guidance](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md), but is generally applicable for any dual-region setup.

Before proceeding with the operational procedure, thoroughly review and understand the contents of the [dual-region concept page](./../../concepts/multi-region/dual-region.md). This page outlines various limitations and requirements pertinent to the procedure, which are crucial for successful execution.

## Disclaimer

:::caution

> Running a dual-region configuration requires the users to detect and manage any regional failures and implement the operational procedure for failover and failback, matching their environments, utilizing the operational procedure below.

Reviewer: ```suggestion
Running a dual-region configuration requires users to detect and manage any regional failures, and implement the operational procedure for failover and failback that matches their environment.
```

I'm not sure we need to double-emphasize the operational procedure here; if anything, we could link to the procedure header?

- **Surviving region**
  - A surviving region refers to a region within a dual-region setup that remains operational and unaffected by a failure or disaster that affects other regions.
- **Lost region**
  - A lost region is a region within a dual-region setup that becomes unavailable or unusable due to a failure or disaster.
- **Recreated region**
  - A recreated region is a region within a dual-region setup that was previously lost but has been restored or recreated to resume its operational state.
  - We assume this region does not contain Camunda 8 deployments or related persistent volumes. Ensure this is the case before executing the failover procedure.

## Procedure

We use the same procedure to handle the loss of both active and passive regions. For clarity, this section focuses on the scenario where the passive region is lost while the active region remains operational. The same procedure will be valid in case of active region loss.

> **Temporary Loss Scenario:** If a region loss is temporary—such as from transient network issues—Zeebe can handle this situation without initiating recovery procedures, provided there is sufficient free space on the persistent disk. However, processing may halt due to a loss of quorum during this time.

Reviewer 2: whitespaces missing between `issues-Zeebe`?

2. **Failover Phase:** Temporarily restores Camunda 8 functionality by removing the lost brokers and handling the export to the unreachable Elasticsearch instance.
3. **Failback Phase:** Fully restores the failed region to its original functionality. This phase requires the region to be ready for the redeployment of Camunda 8.

:::caution

For the failback procedure, the recreated region must not include any active Camunda 8 deployments or residual persistent volumes associated with Camunda 8 or its Elasticsearch instance. It is essential to initiate a clean deployment to prevent data replication and state conflicts.

:::

### Prerequisites

The following procedures assume that the dual-region deployment has been created using [AWS setup guide](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md#deploy-camunda-8-to-the-clusters). We assume you have your own copy of the [c8-multi-region](https://github.com/camunda/c8-multi-region) repository and previously completed changes in the `camunda-values.yml` to adjust them in your setup.

> Ensure you have followed [deploy Camunda 8 to the clusters](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md#deploy-camunda-8-to-the-clusters) to have Camunda 8 installed and configured for a dual-region setup and have the general environment variables (see [environment prerequisites](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md#environment-prerequisites) already set up.

Reviewer: ```suggestion
Follow the [dual-region cluster deployment](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md#deploy-camunda-8-to-the-clusters) guide to install Camunda 8, configure a dual-region setup, and have the general environment variables (see [environment prerequisites](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md#environment-prerequisites) already set up.
```

| You have ensured that you fully lost a region and want to start the temporary recovery. <br /> <br /> One of the regions is lost, meaning Zeebe: <br /> - No data has been lost thanks to Zeebe data replication. <br /> - Is unable to process new requests due to losing the quorum <br /> - Stops exporting new data to Elasticsearch in the lost region <br /> - Stops exporting new data to Elasticsearch in the survived region | The lost brokers have been removed from the Zeebe cluster. <br /> <br /> Continued processing is enabled, and new brokers in the failback procedure will only join the cluster with our intervention. |

#### Procedure

Start with creating a port-forward to the `Zeebe Gateway` in the surviving region to the local host to interact with the Gateway.

The following alternatives to port-forwarding are possible:

> - if the gateway is not exposed to the outside of the Kubernetes cluster, you can skip port-forwarding and use the URL directly

Reviewer 2: the contrary is the case

if exposed --> use URL
if not --> port-forwarding

not exposed --> can't use URL

| **Camunda 8**            | A standalone region with a fully functional Camunda 8 setup, including Zeebe, Operate, Tasklist, and Elasticsearch. | Restore dual-region functionality by deploying Camunda 8 (Zeebe and Elasticsearch) to the newly restored region. |
| **Operate and Tasklist** | Operate and Tasklist are operational in the standalone region.                                                      | Operate and Tasklist need to stay disabled to avoid interference during the database backup and restore process. |

#### Procedure

> This procedure requires your Helm values file, `camunda-values.yml,` in `aws/dual-region/kubernetes,` used to deploy Dial-region Camunda clusters.

Reviewer 2: @theburi the `dial` region remains ☎️.
The previous comment was about the spelling not the uppercase.
`dual-region`

Ensure that the values for `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCHREGION0_ARGS_URL` and `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCHREGION1_ARGS_URL` correctly point to their respective regions. The placeholder in `ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS` should contain the Zeebe endpoints for both regions, the result of the `aws/dual-region/scripts/generate_zeebe_helm_values.sh`.

> Additionally, execute the following Helm command to disable Operate and Tasklist. These components will only be enabled at the end of the region recovery. Keeping them disabled in the newly created region is necessary to avoid data duplication by their Elasticsearch importers.

Reviewer 2: just a side-note, no actual change - not only that, Elastic wouldn't be happy about pre-existing indices.

#### Verification

> The following command will show the pods deployed in the newly created region.

Reviewer: Should we move the actual command beneath this? It becomes less clear that the command a few lines down is related.

Author: Done

> Depending on your chosen `clusterSize`, you should see that half of the amount is spawned in Zeebe brokers.

Reviewer: ```suggestion
Half of the amount of your set `clusterSize` is used to spawn Zeebe brokers.
```
^ This could potentially be included with the example line below. I'm unsure with how it's currently written if this suggestion is what's meant, so shortening it up to be clear may help.

The following command will show the deployed pods of the newly created region.

> Depending on your chosen `clusters`, you should see that half of the amount is spawned in Zeebe brokers.

Reviewer 2: I'd keep it `clusterSize` as it refers to the below value in the Zeebe settings in line 427.
Otherwise, `clusters` is such an overused word in our context - Kubernetes cluster, Zeebe Cluster, Camunda cluster - that it won't be immediately visible what is actually being referred to.

:::note

> This step **does not** affect the process instances in any way. The impact is that process information about the affected instances might not be visible in the Operate and Tasklist.

Reviewer: Agreed here, if the article is to specify operate/tasklist in a specific region, we should clarify the region. 

```suggestion
This step **does not** affect the process instances in any way. Process information may not be visible in Operate and Tasklist running in the affected instance.
```

  <summary>Example output</summary>
  <summary>

> The important part is the `state: "SUCCESS"` and that properties `done` and `total` have equal values. **This is only an example and the values will differ for you.**

Reviewer: ```suggestion
**This is only an example, and the values will differ for you.** Ensure you see `state: "SUCCESS"`, and that the properties `done` and `total` have equal values. 
```

| **Camunda 8**            | Remain unreachable by end-users while restoring functionality. | Enable Operate and Tasklist in both the surviving and recreated regions to allow user interaction with Camunda 8 again. |
| **Elasticsearch Backup** | Backup has been created and restored to the recreated region.  | N/A                                                                                                                     |

#### Procedure

> The base Helm values file `camunda-values.yml` in `aws/dual-region/Kubernetes` contains the adjustments for Elasticsearch and the Zeebe initial brokers. This means we just have to reapply/upgrade the Helm release to enable and deploy Operate and Tasklist.

Reviewer 2: the folder is called `kubernetes`, please don't

</div>
  </TabItem>
</Tabs>

> In conclusion, adhering to this updated operational procedure ensures a structured and efficient recovery process for maintaining operational continuity in dual-region deployments. Please remain caution in managing dual-region environments and be prepared to implement the outlined steps for successful failover and failback.

Reviewer 2: I'd propose changing the wording as it sounds strange:
> Please remain caution in managing ...

--> 

> Please remain cautious when managing ...
